### PR TITLE
Fix sort node deserialization from proto

### PR DESCRIPTION
**Pull request by palaska**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

**Review by andygrove — APPROVED** [review]
Good catch. Thanks @palaska

File: datafusion/proto/src/logical_plan/mod.rs [hunk 1]
```diff
@@ -490,7 +490,14 @@ impl AsLogicalPlan for LogicalPlanNode {
                     into_logical_plan!(sort.input, ctx, extension_codec)?;
                 let sort_expr: Vec<SortExpr> =
                     from_proto::parse_sorts(&sort.expr, ctx, extension_codec)?;
-                LogicalPlanBuilder::from(input).sort(sort_expr)?.build()
+                let fetch: usize = sort.fetch.try_into().map_err(|_| {
+                    DataFusionError::Internal(String::from(
+                        "Protobuf deserialization error, invalid limit value'",
```
Comment (andygrove) [inline datafusion/proto/src/logical_plan/mod.rs:495]: It would be good to show the invalid value and/or the error message to ease debugging

**Comment by andygrove** [comment]
Also, could you add/update a unit test so that we prevent future regressions?

**Review by alamb — COMMENTED** [review]
Thanks @palaska and @andygrove

**Comment by alamb** [comment]
I took the liberty of fixing the `clippy` error and pushing the fix to this branch to get CI to pass (example of [failing CI](https://github.com/apache/datafusion/actions/runs/11039247464/job/30729272038?pr=12626))

**Comment by palaska** [comment]
I guess when no `LIMIT` is specified with `ORDER BY`, our default behavior was to fallback to `-1` since `fetch` is not an optional arg in our proto definition. I made the conversion into usize best effort so things should work as expected now

File: datafusion/proto/tests/cases/roundtrip_logical_plan.rs [hunk 2]
```diff
@@ -704,6 +730,7 @@ async fn roundtrip_logical_plan_distinct_on() -> Result<()> {
     let plan = ctx.sql(query).await?.into_optimized_plan()?;
 
     let bytes = logical_plan_to_bytes(&plan)?;
+    println!("plan {:?}", plan);
```
Comment (andygrove) [inline datafusion/proto/tests/cases/roundtrip_logical_plan.rs:733]: nit: could you remove this debug println?

Comment (palaska) [reply datafusion/proto/tests/cases/roundtrip_logical_plan.rs]: oops

**Pull request merged**